extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java

```diff
@@ -92,11 +88,15 @@ public QuartzScheduler(SchedulerContext context, QuartzSupport quartzSupport, Co
                 for (ScheduledMethodMetadata method : context.getScheduledMethods()) {
 
                     invokers.put(method.getInvokerClassName(), context.createInvoker(method.getInvokerClassName()));
+                    int nameSequence = 0;
 
                     for (Scheduled scheduled : method.getSchedules()) {
-                        String name = triggerNameSequence.getAndIncrement() + "_" + method.getInvokerClassName();
+                        String identity = scheduled.identity().trim();
+                        if (identity.isEmpty()) {
+                            identity = ++nameSequence + "_" + method.getInvokerClassName();
```

Comment:
I'm not sure I see how this solves the issue?

Reply:
Because of the way beans are registered (in an unordered data structure), the list of scheduled methods (invoked methods) did not guarantee that the order is respected in between builds. So instead of having a global sequence strategy for the whole list - whose order can change in between builds, we just maintain a separate sequence for each method which does not change unless otherwise multiple `@Scheduled` annotations are re-arranged (source code change) for a given scheduled method which won't be a problem because of - https://github.com/quarkusio/quarkus/blob/master/extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java#L150-L152.

Reply:
OK. An important thing is: https://stackoverflow.com/questions/26086538/order-of-automatically-wrapped-repeatable-annotations .